airbyte-integrations/connectors/source-airtable/source_airtable/source.py

```diff
@@ -0,0 +1,131 @@
+            processed_record.update(data)
+            processed_records.append(processed_record)
+        return processed_records
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        json_response = response.json()
+        records = json_response.get("records", [])
+        records = self.process_records(records)
+        yield from records
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return f"{self.base_id}/{self.table_name}"
+
+
+# Source
+class SourceAirtable(AbstractSource):
+    schemas = {}
+
+    def check_connection(self, logger, config) -> Tuple[bool, any]:
+        auth = TokenAuthenticator(token=config["api_key"]).get_auth_header()
+        for table in config["tables"]:
+            url = f"https://api.airtable.com/v0/{config['base_id']}/{table}?pageSize=1"
+            try:
+                response = requests.get(url, headers=auth)
+                response.raise_for_status()
+            except requests.exceptions.HTTPError as e:
+                return False, e
+        return True, None
+
+    def discover(self, logger: AirbyteLogger, config) -> AirbyteCatalog:
+        streams = []
+        auth = TokenAuthenticator(token=config["api_key"]).get_auth_header()
+        for table in config["tables"]:
+            url = f"https://api.airtable.com/v0/{config['base_id']}/{table}?pageSize=1"
+            response = requests.get(url, headers=auth)
+            response.raise_for_status()
+            record = response.json().get("records", [])[0].get("fields", {})
+            properties = {
+                "_airtable_id": {"type": ["null", "string"]},
+                "_airtable_created_time": {"type": ["null", "string"]},
+            }
+
+            for field in record:
+                properties[field] = {"type": ["null", "string"]}
+                json_schema = {
+                    "$schema": "http://json-schema.org/draft-07/schema#",
+                    "type": "object",
+                    "properties": properties,
+                }
+
+            self.schemas[table] = json_schema
+            streams.append(
+                AirbyteStream(
+                    name=table,
+                    json_schema=json_schema,
+                    supported_sync_modes=[SyncMode.full_refresh],
+                    supported_destination_sync_modes=[DestinationSyncMode.overwrite, DestinationSyncMode.append_dedup],
+                )
+            )
+        return AirbyteCatalog(streams=streams)
+
+    def streams(self, config: Mapping[str, Any]) -> List[Stream]:
+        auth = TokenAuthenticator(token=config["api_key"])
+        streams = []
+        for table_name in config["tables"]:
+            kwargs = {"base_id": config["base_id"], "table_name": table_name, "authenticator": auth, "schema": self.schemas.get(table_name)}
```

Comment:
I'm not really comfortable with the fact  that `streams` function depends of a previous `discover` call to set `self.schemas`.

Reply:
I have refactored and used helper functions for this